cmd/cosign/cli/options/key.go

```diff
@@ -36,6 +36,9 @@ type KeyOpts struct {
 	TSAServerURL         string
 	RFC3161TimestampPath string
 	TSACertChainPath     string
+	// IssueCertificate controls whether to issue a certificate when a key is
+	// provided.
+	IssueCertificate bool
```

Comment:
IMO this is a misleading name in the context of key options. We will often issue a certificate even when it is false!

Maybe we should give it a better name? `IssueCertificateForExistingKey`? (`IssueCertificate` is fine for the CLI options, though).

Then, you could add an `IssueCertificate()` method that basically returns `ko.IssueCertificateForExistingKey || ko.Sk || ko.KeyRef != ""`. And then call *that* when deciding whether to call out to Fulcio.